src/app/server/Dnssd.cpp

```diff
@@ -453,7 +466,7 @@ void DnssdServer::StartServer(chip::Dnssd::CommissioningMode mode)
         }
 #endif // CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY
     }
-    else
+    else if (!mode.HasValue())
```

Comment:
I'm struggling to understand what happens re: the final `else` that was there. Is it really possible to reach here and have `mode.HasValue()` ? If so, why not have an else?

Reply:
Oh, good catch.  Yes, it's possible to reach here with `mode.HasValue()`.  I need to fix this!

Reply:
For the record this was not breaking CI because the Linux version of all-clusters app does the following, in order:
1. `StartServer(kEnabledBasic)` via `OpenBasicCommissioningWindow` called from `Server::Init`, because `CHIP_DEVICE_CONFIG_ENABLE_PAIRING_AUTOSTART`.
2. `StartServer()` (no args) called from `Server::Init` because
    ```
    CHIP_DEVICE_CONFIG_ENABLE_DNSSD && !CHIP_DEVICE_LAYER_TARGET_ESP32 && !CHIP_DEVICE_LAYER_TARGET_MBED && \
        	    (!CHIP_DEVICE_LAYER_TARGET_AMEBA || !CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE)
    ```

I don't understand why we are doing that, exactly.  And the comments for that condition on that second call suggest that the excluded platforms are excluded only because they do their own separate no-arg `StartServer` calls...

So the fact that the `kEnabledBasic` from the first call was being ignored (incorrectly) was hidden by that second call.

I wish we had a way to query what the server is advertising right now so we could write a unit test for this...